### PR TITLE
Add `sign_and_send_swap_transaction` API to wallet 

### PR DESCRIPTION
**Pull request by thomaseizinger**

### CODE REVIEW

**Comment by thomaseizinger** [comment]
The fuzzer failure is dealt with in https://github.com/comit-network/droplet/pull/86.

File: elements-fun/src/transaction.rs [hunk 1]
```diff
@@ -327,6 +327,30 @@ pub struct PegoutData<'txo> {
     pub extra_data: Vec<&'txo [u8]>,
 }
 
+/// These constants have been reverse engineered through the following transactions:
+///
+/// https://blockstream.info/liquid/tx/a17f4063b3a5fdf46a7012c82390a337e9a0f921933dccfb8a40241b828702f2
+/// https://blockstream.info/liquid/tx/d12ff4e851816908810c7abc839dd5da2c54ad24b4b52800187bee47df96dd5c
+/// https://blockstream.info/liquid/tx/47e60a3bc5beed45a2cf9fb7a8d8969bab4121df98b0034fb0d44f6ed2d60c7d
```
Comment (bonomat) [inline elements-fun/src/transaction.rs:334]: Did you see that these 3 are `p2sh` transactions? 

An improvement could be to do 3 transactions with the mentioned combinations and record the vbytes.

**Review by bonomat — APPROVED** [review]
Beautiful commit messages.

File: elements-fun/src/transaction.rs (hunk 1 above)
Comment (thomaseizinger) [reply elements-fun/src/transaction.rs:334]: Note: This code was only moved in this PR, not added.

The majority of the size comes from the range proof which is almost a kilobyte. I'd expect the size of the script to therefore play a negligible role.

File: bobtimus/src/lib.rs [hunk 2]
```diff
@@ -154,40 +148,52 @@ impl<R, RS> Bobtimus<R, RS> {
...
+        )?;
 
-        let message = bob1.compose(signer).await?;
+        let transaction = swap::bob_create_transaction(
+            &mut self.rng,
+            &self.secp,
+            alice,
+            bob,
+            self.btc_asset_id,
+            Amount::from_sat(1), // TODO: Make this dynamic once there is something going on on Liquid
```
[the remark this answers is not in the document]
Comment (thomaseizinger) [reply bobtimus/src/lib.rs:184]: I know about the minimum but unfortunately we can't express it with `Amount`.

File: bobtimus/src/lib.rs [hunk 3]
```diff
@@ -154,40 +148,52 @@ impl<R, RS> Bobtimus<R, RS> {
                             })?
                             .clone();
 
-                        Result::<_, anyhow::Error>::Ok((txin, txin_as_txout, blinding_key))
+                        Result::<_, anyhow::Error>::Ok(swap::Input {
+                            tx_in,
+                            tx_out,
```
Comment (luckysori) [inline bobtimus/src/lib.rs:153]: :upside_down_face: Super nitpicky, but we almost exclusively use `txin` and `txout` in this workspace.

File: swap/src/lib.rs [hunk 4]
```diff
@@ -1,462 +1,307 @@
...
-            let sig = secp.sign(&Message::from(sighash), &input_sk);
-
-            let mut serialized_signature = sig.serialize_der().to_vec();
-            serialized_signature.push(SigHashType::All as u8);
-
-            vec![serialized_signature, input_pk_bob.serialize().to_vec()]
-        };
+    pub fn calculate_change_amount(
+        &self,
+        other_receive_asset: AssetId,
```
Comment (luckysori) [inline swap/src/lib.rs:268]: :question: Maybe `send_asset`?

**Review by luckysori — APPROVED** [review]
Really nice effort. I wholeheartedly agree with the TODO indicating that we should replace the `swap` code with PSET asap.

File: swap/src/lib.rs (hunk 4 above)
Comment (thomaseizinger) [reply swap/src/lib.rs:268]: I am indifferent and lazy so I will just leave it at that :D

**Pull request merged**